### PR TITLE
Update dev setup to support modern JS syntax

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Change List
- eslint config
- Fix a bug in jupyter-widget/playground where `babel-plugin-inline-webgl-constants` removes import
- Fix `test bench` failure
- Add proof-of-concept (optional chaining) in test code

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -22,13 +22,14 @@ module.exports = {
   rules: {
     'guard-for-in': 0,
     'no-inline-comments': 0,
+    'no-unused-expressions': 0,
```
Comment (ibgreen) [inline .eslintrc.js:25]: This seems unnecessary. At least a comment explaining why we are adding this?

File: test/bench/node.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
-require('reify');
+require('@babel/register');
```
Comment (ibgreen) [inline test/bench/node.js:1]: Noooo not babel register..., please let me debug untranspiled code.... ;)

Just bump to node 14 to ecma script 2020 support.

File: test/node.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-require('reify');
+require('@babel/register');
```
Comment (ibgreen) [inline test/node.js:1]: Ditto

**Review by ibgreen — APPROVED** [review]

**Review by alasarr — APPROVED** [review]

File: test/bench/node.js (hunk 2 above)
Comment (Pessimistress) [reply test/bench/node.js:1]: Node.js does not support our style of module imports (a mix of es modules and commonjs).

`reify` was meant to add that support, however it crashes on ECMA 2020 syntax. The module has also not been updated for 2 years.

I have added an explicit option to `@babel/register` to only use transforms that are needed by Node 14.

Comment (ibgreen) [reply test/bench/node.js]: Got it, that is not too bad. FWIW, there is `esm` too (reify alternative) it may be more up to date.

**Pull request merged**